app/lib/foreman_deployments/config/array.rb

```diff
@@ -0,0 +1,83 @@
+module ForemanDeployments
+  module Config
+    class Array < ::Array
+      def configure(values)
+        values = ensure_hash(values)
+        values.map do |index, value|
+          configure_index(index, value)
+        end
+      end
+
+      def configured
```

Comment:
I would like to hide the configuration process from the user. I think that the writer of the stack should not be aware to the "configurable" nature of the input parameters.
IMHO it would be nicer if the regular array methods will expose the final result. This way we will only need to store indexes of values that came from configuration, in order to recreate the original array.